integration/tests/monitor/rate-limiter.test.js

```diff
@@ -0,0 +1,23 @@
+const assert = require('node:assert/strict');
+const { describe, it } = require("node:test");
+
+const RateLimiter = require("../../monitor/rate-limiter");
+
+describe("lease()", () => {
+  it("rejects a lease within the window", () => {
+    const rateLimiter = new RateLimiter(1000, 1);
+
+    assert.equal(rateLimiter.lease(), true);
+    assert.equal(rateLimiter.lease(), false);
+  });
```

Comment:
📓 This probably isn't the optimal way to test this. There was a lot of movement around testing between Node 18 (our version) and 21 (the latest version). We should come back and try to mock the timeout when we've upgraded.